source/code/plugins/in_dsc_monitor.rb

```diff
@@ -0,0 +1,112 @@
+require 'fluent/input'
+require 'fluent/config/error'
+require 'yaml/store'
+
+module Fluent
+  class DscMonitoringInput < Input
+    Fluent::Plugin.register_input('dsc_monitor', self)
+
+    config_param :tag, :string, :default=>nil
+    config_param :check_install_interval, :time, :default=>86400
+    config_param :check_status_interval, :time, :default=>1800
+    config_param :dsc_cache_file, :string, :default=>'/var/opt/microsoft/omsagent/state/dsc_cache.yml'
+   
+    def configure(conf)
+      super
+      if !@tag 
+        raise Fluent::ConfigError, "'tag' option is required on dsc_checks input"
+      end
+    end
+
+    def start
+      super
+      @finished_check_install = false
+      @finished_check_status = false
+      @thread_check_install = Thread.new(&method(:run_check_install))
+      @thread_check_status = Thread.new(&method(:run_check_status))
+      @dsc_cache = YAML::Store.new(dsc_cache_file) #Creates the file if it doesn't exist; otherwise, the existing file will be read.
+    end
+
+    def check_install
+      dpkg = (%x(which dpkg > /dev/null 2>&1; echo $?)).to_i
+      if dpkg == 0
+        %x(dpkg --list omsconfig > /dev/null 2>&1; echo $?).to_i
+      else
+        %x(rpm -qi omsconfig > /dev/null 2>&1; echo $?).to_i
+      end
+    end
+
+    def run_check_install
+      until @finished_check_install
+        install_status = check_install
+
+        # store dsc installation status in the cache
+        @dsc_cache.transaction {
```

Comment:
you don't need to cache the install status. you need to get it every time anyway.